tests/diffs/test_resolving.py

```diff
@@ -0,0 +1,28 @@
+import pytest
+
+from kopf.structs.diffs import resolve
+
+
+def test_existing_key():
+    d = {'abc': {'def': {'hij': 'val'}}}
+    r = resolve(d, ['abc', 'def', 'hij'])
+    assert r == 'val'
+
+
+def test_unexisting_key():
+    d = {'abc': {'def': {'hij': 'val'}}}
+    with pytest.raises(KeyError):
+        resolve(d, ['abc', 'def', 'xyz'])
+
+
+def test_nonmapping_key():
+    d = {'key': 'val'}
+    with pytest.raises(TypeError):
+        resolve(d, ['key', 'sub'])
```

Comment:
```suggestion
        resolve(d, ['key', 'val'])
```
I understand from non mapping that the key must be correct though

Reply:
The unexistent key is tested in `test_unexisting_key` above. There, the data structure before failure is a mapping `d['abc']['def']` equal to `{'hij': 'val'}`, and we tried `{'hij': 'val'}['xyz']` — it raises `KeyError`.

Here, we use the actually existing key — `d['key']` — which has a value that is not a mapping: `"val"`. It is not possible to do `"val"['key']` — it raises `TypeError`. 

I can probably align this test with the previous one so that they look the same, and then I can select `['abc']['def']['hij']['key']` — with the same effect.